new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1258 @@
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+  }
+
+  /**
+   * Verify all server pods are running.
+   * Verify k8s services for all servers are created.
+   */
+  @BeforeEach
+  public void beforeEach() {
+
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
+
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check managed server service {0} is created in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkServiceExists(managedServerPrefix + i, domainNamespace);
+    }
+
+    // Check managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server pod {0} to be ready in namespace {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+
+    // Check configured cluster configuration is available 
+    boolean isServerConfigured = 
+        checkManagedServerConfiguration("config-cluster-server1");
+    assertTrue(isServerConfigured, 
+        "Could not find managed server from configured cluster");
+    logger.info("Found managed server from configured cluster");
+
+    // Check standalone server configuration is available 
+    isServerConfigured = 
+       checkManagedServerConfiguration("standalone-managed");
+    logger.info("Found non-cluster managed server configuration");
+  }
+
+  /**
+   * Stop the Administration server by patching the resource definition with 
+   *  spec/adminServer/serverStartPolicy set to NEVER.
+   * Make sure that Only the Administration server is stopped. 
+   * Restart the Administration server by patching the resource definition with 
+   *  spec/adminServer/serverStartPolicy set to IF_NEEDED.
+   * Make sure that the Administration server is in RUNNING state. 
+   */
+  @Test
+  @DisplayName("Restart the Administration server with serverStartPolicy")
+  public void testAdminServerRestart() {
+
+    String configServerPodName = domainUid + "-config-cluster-server1";
+    String dynamicServerPodName = domainUid + "-managed-server1";
+    DateTime dynTs = getPodCreationTime(domainNamespace, dynamicServerPodName);
+    DateTime cfgTs = getPodCreationTime(domainNamespace, configServerPodName);
+    StringBuffer patchStr = null;
```

Comment:
lot of duplicated code, create a private method which takes path, value, other needed vars and patches the domain

Reply:
done